oneclick.py

```diff
@@ -0,0 +1,600 @@
+    DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ" #ex: 2013-12-27T00:00:00Z
+    DATE_FORMAT = "%Y-%m-%d" #ex: 2013-12-27T00:00:00Z
+
+    log = logging.getLogger("OneClick representation extractor")
+
+    # TODO: the formats/encoding/drm information needs confirmation from OneClick
+    oneclick_formats = {
+        "ebook-epub-oneclick" : (
+            Representation.EPUB_MEDIA_TYPE, DeliveryMechanism.ONECLICK_DRM
+        ),
+        "ebook-epub-open" : (
+            Representation.EPUB_MEDIA_TYPE, DeliveryMechanism.NO_DRM
+        ),
+        "audiobook-mp3-oneclick" : (
+            "application/x-od-media", DeliveryMechanism.ONECLICK_DRM
+        ),
+    }
+
+    oneclick_medium_to_simplified_medium = {
+        "eBook" : Edition.BOOK_MEDIUM,
+        "eAudio" : Edition.AUDIO_MEDIUM,
+    }
+
+
+    @classmethod
+    def image_link_to_linkdata(cls, link_url, rel):
+        if not link_url or (link_url.find("http") < 0):
+            return None
+
+        media_type = None
+        if link_url.endswith(".jpg"):
+            media_type = "image/jpeg"
+
+        return LinkData(rel=rel, href=link_url, media_type=media_type)
+
+
+    @classmethod
+    def isbn_info_to_metadata(cls, book, include_bibliographic=True, include_formats=True):
+        """Turn OneClick's JSON representation of a book into a Metadata object.
+        Assumes the JSON is in the format that comes from the media/{isbn} endpoint.
+
+        :param book a json response-derived dictionary of book attributes
+        """
+        if not 'isbn' in book:
+            return None
+        oneclick_id = book['isbn']
+        primary_identifier = IdentifierData(
+            Identifier.ONECLICK_ID, oneclick_id
+        )
+
+        metadata = Metadata(
+            data_source=DataSource.ONECLICK,
+            primary_identifier=primary_identifier,
+        )
+
+        if include_bibliographic:
+            title = book.get('title', None)
+            series_name = book.get('seriesName', None)
+            # ignored for now
+            series_position = book.get('seriesPosition', None)
+            # ignored for now
+            series_total = book.get('seriesTotal', None)
+            # ignored for now
+            has_digital_rights = book.get('hasDigitalRights', None)
+
+            publisher = book.get('publisher', None)
+            if 'publicationDate' in book:
+                published = datetime.datetime.strptime(
+                    book['publicationDate'][:10], cls.DATE_FORMAT)
+            else:
+                published = None
+
+            if 'language' in book:
+                language = LanguageCodes.string_to_alpha_3(book['language'])
+            else:
+                language = 'eng'
+
+            contributors = []
+            if 'authors' in book:
+                authors = book['authors']
+                for author in authors.split(";"):
+                    sort_name = author.strip()
+                    roles = [Contributor.AUTHOR_ROLE]
```

Comment:
OK, great. I know Bibliotheca treats all roles as author roles. One Click might have made a similar mistake but it's less likely since they handle narrators specially.